File: lib/common_validations.rb
module CommonValidations
  BASE_HOST_FORMAT = %r{[a-zA-Z0-9:][a-zA-Z0-9\-\.:]*}
  HOST_FORMAT = %r{^#{BASE_HOST_FORMAT.source}$}
  HOST_FORMAT_WITH_WILDCARD = %r{^(\*|\*?#{BASE_HOST_FORMAT.source}|\*\.#{BASE_HOST_FORMAT.source})$}
  URL_PREFIX_FORMAT = %r{^/}
end

File: src/api-umbrella/web-app/utils/common_validations.lua
local base_host_format = [[[a-zA-Z0-9:][a-zA-Z0-9\-\.:]*]]

return {
  base_host_format = base_host_format,
  host_format = "^" .. base_host_format .. "$",
  host_format_with_wildcard = [[^(\*|(\*\.|\.)]] .. base_host_format .. "|" .. base_host_format .. ")$",
  url_prefix_format = "^/",
  uuid = [[^[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}$]],
}
